### ISSUE TITLE
[BUG] Inconsistency in generateOptions

### ISSUE THREAD

**Issue by josvanroosmalen** [marklogic/marklogic-data-hub#3434]
The generateOptions need to be changed. In case of an Exception we only print a stacktrace. Then we return "{}". This is problematic because the entity options are (sometimes) XML.  It should not print a stacktrace, but it should rethrow a RuntimeException. 

```
  String generateOptions(List<JsonNode> entities, boolean forExplorer) {
            try {
                ObjectMapper objectMapper = new ObjectMapper();
                JsonNode node = objectMapper.valueToTree(entities);
                RequestParameters params = new RequestParameters();
                params.put("forExplorer", Boolean.toString(forExplorer));

                ResourceServices.ServiceResultIterator resultItr = this.getServices().post(params, new JacksonHandle(node));
                if (resultItr == null || !resultItr.hasNext()) {
                    throw new IOException("Unable to generate query options");
                }
                ResourceServices.ServiceResult res = resultItr.next();
                return res.getContent(new StringHandle()).get();
            } catch (Exception e) {
                e.printStackTrace();
            }
            return "{}";
        }
    }
```

**Comment by rjrudin** [comment]
What exception did you run into, and how did it occur?

**Comment by rjrudin** [comment]
Opened https://project.marklogic.com/jira/browse/DHFPROD-4071 . But let us know what error you ran into, and whether you were able to fix it yourself.

**Comment by kkanthet** [comment]
Fixed as part of 5.2.0 release
